services/swarm/client/client.go

```diff
@@ -0,0 +1,204 @@
+package client
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+	"sync/atomic"
+)
+
+type Config struct {
+	URLs       []string
+	APIVersion string
+	TLSConfig  *tls.Config
+}
+
+type Client interface {
+	Scales(name string) ScalesInterface
+	Update(c Config) error
+}
```

Comment:
With Kubernetes and Swarm.
"Get" call for "reading the name of k8s resource name or docker swarm service name" remains the same logic .i.e just make a "Get" API call. 
"Update" call for updating the k8s resource for scaling or updating the docker swarm service to scale remains same logic .i.e just make a "POST" API call.

However, K8 has a specific implementation for autoscaling whereas docker swarm doesn't have autoscaling as of current available version and only supports scaling the containers(no autoscaling). So, w.r.t to Swarm we have to get the swarm service object and update it with "replicas" with the desired number.   

Though the "Get" and "Update" logic remains the same and also there is some space to merge the backends, I would prefer separate implementation for k8 and swarm as they have different API Specs. 

I am new to Kapacitor and also to golang. So, this just my view.
You guys can take the decision.